### ISSUE TITLE
.NET SDK throws exception when GetCheckAsync returns allowed: false

### ISSUE THREAD

**Issue by Dakraeth** [ory/keto#851]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/keto/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/keto/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Cloud](https://www.ory.sh/) project.
- [X] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [X] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe the bug

When using GetCheckAsync of the .NET Keto SDK, an exception is thrown for requests that are not allowed.

### Reproducing the bug

I have created a repository to reproduce this behavior. [Github Repository](https://github.com/Dakraeth/ory-keto-getcheck-issue)

1. clone repository
2. start docker containers
`docker-compose up -d`
3. start .NET api
`dotnet run`
4. open webbrowser [api swagger url](https://localhost:7186) (you may need to change the port)
5. on swagger hit the CreateExampleRelationTuple endpoint to create an example RealationTuple (or call endpoint directly via [endpoint](https://localhost:7186/OryKeto/CreateExampleRelationTuple))
6. use GetCheckAsyncSuccess for an working success example(or call endpoint directly via [endpoint](https://localhost:7186/OryKeto/GetCheckAsyncSuccess))
7. use GetCheckAsyncFailure for an example with an exception(or call endpoint directly via [endpoint](https://localhost:7186/OryKeto/GetCheckAsyncFailure))
8. use GetCheckAsyncWorkaround for an workaround (or call endpoint directly via [endpoint](https://localhost:7186/OryKeto/GetCheckAsyncWorkaround))

### Version

oryd/keto:v0.8.0

### On which operating system are you observing this issue?

Linux

### In which environment are you deploying?

Docker Compose

**Comment by zepatrik** [comment]
Thanks for this. As pointed out previously, the status code in case of a false check response is set to `401` on purpose to allow easy integration with other services. Maybe @aeneasr has an idea on how this can be improved in the SDKs? Maybe we can add a second simple endpoint that does not set the status code?
This likely also affects other SDKs.

Current workaround: catch the `401` exception and treat it as a not allowed response.

**Comment by aeneasr** [comment]
I believe that error messages will always return an error response in OpenAPI generated clients. You can type those for better assertiveness, but it will still throw an error!

**Comment by zepatrik** [comment]
It is not an error message, it is a legit response:
https://github.com/ory/keto/blob/ef103eb231c6ca466d7c928ca73b29bf3d4c23d1/internal/check/handler.go#L160-L165

**Comment by aeneasr** [comment]
Sorry, I meant error status codes. So everything >= 400

**Comment by zepatrik** [comment]
Yes, that is why we should not return an error status code for the SDKs. Not sure what options we have to do that. Do the SDKs set a specific user agent? Or should we add a second route? We could also make it depend on a query parameter.

**Comment by aeneasr** [comment]
I'd suggest to make this as explicit. Interpreting 200 as "ok this passed" even though it didnt due to some obscure User-Agent behavior or other "auto-detect" feature (even query string) would be quite bad. Actually, I think we need to choose for this endpoint whether to walk back and go with the 200 route again.

**Comment by zepatrik** [comment]
OK so having two endpoints makes more sense right?
